// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

// Parameters for cochlear auditory feature extractor.

syntax = "proto3";

import "eidos/audition/auditory_pipeline_config.proto";
import "eidos/audition/stimulus_config.proto";

package eidos.audition;

// Interface for controlling the audio feature extractor.
//
// Next available ID: 3
message AuditoryFeatureExtractorConfig {
  // Stimulus configuration containing input-specific parameters such as
  // sampling rate, model-agnostic analysis parameters and so on.
  StimulusConfig config = 1;

  // Auditory pipeline consisting of one or more models.
  AuditoryPipelineConfig pipeline = 2;
}

// Local Variables:
// mode: c++
// End:
